DAOS-2240 Build CaRT into an RPM with RPM components

brianjmurrell opened the pull request.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/4/display/redirect

daosbuild1 commented:
Test stage Build on CentOS 7 with Clang completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/4/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build on CentOS 7 with Clang completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/6/display/redirect

ashleypittman commented on Jenkinsfile:
```diff
@@ -38,7 +38,7 @@
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
-//@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@pipeline-pr-test-rpms") _
```
Presumably this won't actually me landed until after the pipeline-lib changes are landed?

brianjmurrell replied:
That's right.  This is exactly the kind of thing that commented out `Library()` call is there for.

ashleypittman commented on Jenkinsfile:
```diff
@@ -109,7 +109,7 @@ pipeline {
                             dir 'utils/docker'
                             label 'docker_runner'
-                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS'
+                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS --build-arg DONT_USE_RPMS=false --build-arg OMPI=ompi-3.0.0rc4-1'
```
In general I find variable names that contain a negative to not be a good idea, although this is not a regression.

brianjmurrell replied:
I generally avoid them myself, but sometimes they are more useful as a double negative.  I will review that in this case the next time I am looking over this patch.

ashleypittman commented on Jenkinsfile:
```diff
@@ -164,7 +164,7 @@ pipeline {
                             dir 'utils/docker'
                             label 'docker_runner'
-                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS'
+                            additionalBuildArgs "-t ${sanitized_JOB_NAME}-centos7 " + '$BUILDARGS --build-arg DONT_USE_RPMS=false --build-arg DONT_USE_RPMS=false OMPI=ompi-3.0.0rc4-1'
```
Putting software versions inside a dockerfile isn't something that's currently done, would it be better to perform this action as part of the build step?

ashleypittman commented on utils/docker/Dockerfile.centos:7:
```diff
@@ -49,26 +49,44 @@ ARG NOBUILD
+done
+
+RUN if $DONT_USE_RPMS; then exit 0; fi;                   \
+OMPI=${OMPI} yum -y install openpa-devel libfabric-devel mercury-devel pmix-devel ompi-devel
```
I have no idea what this does?

If it's installing from the repo added earlier do we want to use --disable-repo=* --enable-repo=<our one>

brianjmurrell replied:
Possibly.  But I generally find that when you are building your own packages is because either a) it's not available in any other repo or b) you need something newer.

I tend to like to let `yum` work things our for itself given all of the knowledge.  If we have a package that is older than what is in another repo we really should be adding version restrictions to our dependency of it so that `yum` with the whole picture will still do the right thing.

ashleypittman commented on utils/docker/Dockerfile.centos:7:
```diff
@@ -49,26 +49,44 @@ ARG NOBUILD
+#RUN yum-config-manager --save --setopt=*_job_daos-stack_job_${ext}_job_master_lastSuccessfulBuild_artifact_artifacts_.gpgcheck=0
+RUN if $DONT_USE_RPMS; then exit 0; fi;                                                                                                                     \
+for ext in openpa libfabric mercury pmix ompi; do                                                                                                                     \
+    yum-config-manager --add-repo=https://build.hpdd.intel.com/job/daos-stack/job/${ext}/job/master/lastSuccessfulBuild/artifact/artifacts/;            \
```
I had no idea we were building RPMs for these packages, when/where was this work done?

brianjmurrell replied:
I've done it over the period of a couple of days as a background task while I wait for builds and tests to complete.

FWIW, CaRT builds in 4 minutes with RPMs installed rather than the 30 or so it takes without them.

ashleypittman replied:
> FWIW, CaRT builds in 4 minutes with RPMs installed rather than the 30 or so it takes without them.

That's great progress!

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/16/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/16/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/16/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/16/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/16/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/18/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/18/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/18/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/18/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/19/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/19/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/19/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/19/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/20/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/20/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/21/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/21/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/22/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/22/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/22/display/redirect

daosbuild1 commented on cart.spec:
```diff
@@ -0,0 +1,81 @@
+BuildRequires:	openssl-devel
+BuildRequires:	libcmocka-devel
+BuildRequires:	libyaml-devel
+#Requires:	
```
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 replied:
(style)  trailing whitespace

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/23/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/23/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/24/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/24/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/25/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/25/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/26/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/26/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/27/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/27/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/28/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/29/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/29/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/29/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/29/display/redirect

daosbuild1 commented on Jenkinsfile:
```diff
@@ -47,28 +47,24 @@ def singleNodeTest(test_mode) {
                            set -e
-                           sudo mkdir -p \$CART_BASE
-                           sudo mount -t nfs \$HOSTNAME:\$PWD \$CART_BASE
+                           find . -type f -print0 | xargs -0 ls -l 
```
(style)  trailing whitespace

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/30/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/31/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/31/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/31/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/31/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/31/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/32/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/32/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/32/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/32/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/32/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/34/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/34/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/34/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/34/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/34/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/36/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/36/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/36/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/36/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/36/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/37/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/37/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/37/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/38/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/38/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/38/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/38/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/38/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/40/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/40/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/40/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/40/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/40/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/41/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/41/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/41/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/41/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/41/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/42/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/42/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/42/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/42/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/42/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/43/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/43/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/43/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/44/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/44/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/44/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/46/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/46/display/redirect

daosbuild1 reviewed (commented):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/47/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:
[test/cart_test_cart_ctl.py:130](https://github.com/daos-stack/cart/blob/93f1e6a76fda0cb8271ec82ef7a4066a802861f8/test/cart_test_cart_ctl.py#L130): (pylint-syntax-error) invalid syntax

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/47/display/redirect

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -122,7 +122,8 @@ def test_cart_ctl_five_nodes(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl list_ctx' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
+                                           os.path.sep + 'cart_ctl' list_ctx' + \
```
(pylint-syntax-error) invalid syntax

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/48/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/48/display/redirect

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -188,7 +192,8 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl disable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -188,7 +192,8 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl disable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
+                                           os.path.sep + 'cart_ctl disable_fi' + \
```
(pylint-line-too-long) Line too long (82/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl list_ctx' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl enable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl enable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
+                                           os.path.sep + 'cart_ctl enable_fi' + \
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -178,7 +181,8 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl set_fi_attr' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -178,7 +181,8 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl set_fi_attr' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
+                                           os.path.sep + 'cart_ctl set_fi_attr' + \
```
(pylint-line-too-long) Line too long (83/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -122,7 +122,8 @@ def test_cart_ctl_five_nodes(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl list_ctx' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX', 'bin')) + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/49/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/49/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/49/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/49/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/49/display/redirect

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -188,7 +192,8 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl disable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -188,7 +192,8 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl disable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
+                                           os.path.sep + 'cart_ctl disable_fi' + \
```
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (82/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl list_ctx' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl enable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -161,15 +162,17 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl enable_fi' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
+                                           os.path.sep + 'cart_ctl enable_fi' + \
```
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (81/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -178,7 +181,8 @@ def test_cart_ctl_one_node(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl set_fi_attr' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -178,7 +181,8 @@ def test_cart_ctl_one_node(self):
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl set_fi_attr' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
+                                           os.path.sep + 'cart_ctl set_fi_attr' + \
```
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (83/80)

daosbuild1 commented on test/cart_test_cart_ctl.py:
```diff
@@ -122,7 +122,8 @@ def test_cart_ctl_five_nodes(self):
         procrtn = self.launch_test(testmsg, '1', self.pass_env, \
                                    cli=client, \
-                                   cli_arg='../bin/cart_ctl list_ctx' + \
+                                   cli_arg=os.path.join(os.getenv('CRT_PREFIX'), 'bin') + \
```
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 replied:
(pylint-line-too-long) Line too long (91/80)

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/50/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/50/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/50/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/50/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/50/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/51/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/51/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/51/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/51/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/51/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/52/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/52/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/52/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/52/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/52/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/53/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/53/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/53/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/53/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/53/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/54/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/54/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/54/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/54/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/54/display/redirect

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +195,13 @@ else
-$CART_BASE/install/Linux/TESTING/$log_base_path" install/Linux/TESTING/
+ssh -i ci_key jenkins@"${nodes[0]}" "set -x
+ls -l /var/tmp
+find $log_base_path -type f"
```
(lint) Note that, unescaped, this expands on the client side. [SC2029]

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +195,13 @@ else
+ls -l /var/tmp
+find $log_base_path -type f"
+
+scp -i ci_key -r jenkins@"${nodes[0]}":\
```
(lint) The double quotes around this do nothing. Remove or escape them. [SC2140]

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/55/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/55/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/55/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/55/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/55/display/redirect

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +198,13 @@ else
+ls -l \"$log_base_path_base\"
+find $log_base_path -type f"
+
+scp -i ci_key -r jenkins@"${nodes[0]}":\
```
(lint) The double quotes around this do nothing. Remove or escape them. [SC2140]

daosbuild1 replied:
(lint) The double quotes around this do nothing. Remove or escape them. [SC2140]

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +198,13 @@ else
-scp -i ci_key -r jenkins@"${nodes[0]}:\
-$CART_BASE/install/Linux/TESTING/$log_base_path" install/Linux/TESTING/
+ssh -i ci_key jenkins@"${nodes[0]}" "set -x
+ls -l \"$log_base_path_base\"
```
(lint) Note that, unescaped, this expands on the client side. [SC2029]

daosbuild1 replied:
(lint) Note that, unescaped, this expands on the client side. [SC2029]

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -193,15 +156,15 @@ elif [ \"$1\" = \"3\" ]; then
 EOF
     rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_three_node.cfg \\
+    python3 test_runner config=\"$log_base_path_base\"/cart_multi_three_node.cfg \\
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -221,7 +184,7 @@ elif [ \"$1\" = \"5\" ]; then
 EOF
     rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_five_node.cfg \\
+    python3 test_runner config=\"$log_base_path_base\"/cart_multi_five_node.cfg \\
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -169,15 +132,15 @@ if [ \"$1\" = \"2\" ]; then
 EOF
     rm -rf $log_base_path/
-    python3 test_runner config=scripts/cart_multi_two_node.cfg \\
+    python3 test_runner config=\"$log_base_path_base\"/cart_multi_two_node.cfg \\
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/56/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/56/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/56/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/56/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/56/display/redirect

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +198,13 @@ else
+ls -l \"$log_base_path_base\"
+find $log_base_path -type f"
+
+scp -i ci_key -r jenkins@"${nodes[0]}":\
```
(lint) The double quotes around this do nothing. Remove or escape them. [SC2140]

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +198,13 @@ else
-scp -i ci_key -r "${nodes[0]}:\
-$CART_BASE/install/Linux/TESTING/$log_base_path" install/Linux/TESTING/
+ssh -i ci_key jenkins@"${nodes[0]}" "set -x
+ls -l \"$log_base_path_base\"
```
(lint) Note that, unescaped, this expands on the client side. [SC2029]

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/57/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/57/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/57/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/57/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/57/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/59/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/59/display/redirect

daosbuild1 commented on multi-node-test.sh:
```diff
@@ -235,8 +201,14 @@ else
+ls -l \"$log_base_path_base\"
+find $log_base_path -type f"
+
+scp -i ci_key -r jenkins@"${nodes[0]}":\
```
(lint) The double quotes around this do nothing. Remove or escape them. [SC2140]

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/60/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/60/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/60/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/60/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/60/display/redirect

brianjmurrell committed:
DAOS-2240 Build CaRT as an RPM

By using RPMs for all of it's dependent components.
Also fix single node test exit status.

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/61/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/61/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/61/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/62/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/62/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/62/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/63/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/63/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/63/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/64/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/64/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/64/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/65/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/65/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/65/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/70/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/70/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/70/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/70/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/70/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/73/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/84/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/84/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/84/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/84/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/85/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/85/display/redirect

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/85/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/86/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/86/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/86/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/86/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/86/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Single-node-valgrind completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/89/display/redirect

daosbuild1 commented:
Test stage Two-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/89/display/redirect

daosbuild1 commented:
Test stage Three-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/89/display/redirect

daosbuild1 commented:
Test stage Five-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/89/display/redirect

daosbuild1 commented:
Test stage Single-node completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/89/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

ashleypittman commented on Jenkinsfile:
```diff
@@ -158,6 +159,40 @@ pipeline {
+                        }
+                    }
+                    steps {
+                        sh '''rm -rf artifacts/
```
Didn't we decide the best way to delete prior builds was to use the following in post:

                        cleanup {
                            dir('artifacts') {
                                deleteDir()
                            }
                        }

brianjmurrell replied:
I don't recall that conversation but that could just be my 50 year-old memory.

But generally speaking I usually like to try to preserve state at the end and clean it up at the start (i.e. of the next run) for the cases where it's useful to inspect the state at the end in the case of failures.

That's not a hard rule for me, so I could be talked into changing to clean up at the end if you really prefer it.

ashleypittman replied:
As always my preference is for consistency, that we decide best practice and apply it.  That's what you had me do for the IOF build step.

ashleypittman commented on Makefile:
```diff
@@ -0,0 +1,62 @@
+NAME        := cart
+VERSION     := 0.0.1
```
Does CaRT currently use a version for shared libraries?  It would be good to tie these together at some point.

brianjmurrell replied:
They are currently .0.0.1.  That's why I set the version here to 0.0.1.

ashleypittman commented on cart.spec:
```diff
@@ -0,0 +1,82 @@
+Name:          cart
+Version:       0.0.1
```
Duplicated with what's in Makefile.

brianjmurrell replied:
Yes.  There is some future work to have the `Makefile` get the version from the specfile.

ashleypittman replied:
True.  One thing about scons, the way we define our SConstruct files they can be imported as python, or just run directly, so exporting the version from a version in the python code is actually really easy.

ashleypittman commented on cart.spec:
```diff
@@ -0,0 +1,82 @@
+
+%build
+# remove rpathing from the build
+find . -name SConscript | xargs sed -i -e '/AppendUnique(RPATH=.*)/d'
```
This is very ugly, far preferable would be to have a --no-rpath option to scons, which would not be difficult to add.

brianjmurrell replied:
Fair enough.  But I'm on a deadline to get RPMs for the whole DAOS stack, so I'd like to defer this for future work.

ashleypittman commented on cart.spec:
```diff
@@ -0,0 +1,82 @@
+scons %{?_smp_mflags}                     \
+      --config=force                      \
+      --update-prereq=all                 \
+      --build-deps=yes                    \
```
What deps is this actually building, this specfile looks to have listed them all as dependencies already.  I think it would be possible to put a comma separated list here if there are a couple that we want built, and that would be much more failsafe against future changes.

brianjmurrell replied:
Again, just one of the "standard" options we have successfully used.

ashleypittman replied:
The risk here is the list of dependances is wrong, or changes somehow and the build ends up unintentionally changing so the CaRT rpm starts also including some of the dependances automatically, and we don't notice.  It would be nice to see what happens without this, and as I say to whitelist ones we actually do need, rather than have scons just fix any omissions

brianjmurrell replied:
Right.  I have a patch to remove `--update-prereq=all` and `--build-deps=yes` that I'm getting ready to push.

ashleypittman commented on cart.spec:
```diff
@@ -0,0 +1,82 @@
+
+scons %{?_smp_mflags}                     \
+      --config=force                      \
+      --update-prereq=all                 \
```
Remind me what this option does and why it's needed.

brianjmurrell replied:
I'm not entirely sure TBH.  It's just one of the number of "standard" options that we currently use successfully to build DAOS and CaRT with `scons` so I'd like to keep it consistent.

ashleypittman replied:
It's a trade-off between flexibility and consistency, for RPMs I'd personally rather it was tightly defined and we had to fix things when it breaks, as at least then we know when something changes.

brianjmurrell replied:
Indeed.  Part of my next patch.

brianjmurrell committed:
Remove a couple of scons build options

And add some Requires

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/cart/job/PR-93/98/display/redirect

brianjmurrell committed:
Remove file before trying to gzip it

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
